mindsdb_native/libs/helpers/text_helpers.py

```diff
@@ -175,6 +186,29 @@ def is_foreign_key(data, column_name, data_subtype, other_potential_subtypes):
         all_uuid_charset = all(set(str(x)).issubset(uuid_charset) for x in data)
         is_uuid = all_uuid_charset and all_same_length
 
+        if all_same_length:
+            str_data = [str(x) for x in data]
+            
+            randomness_per_index = []
+            for i, _ in enumerate(str_data[0]):
+                all_ascii = all(isascii(x[i]) for x in str_data)
+                all_alpha = all_ascii and all(x[i].isalpha() for x in str_data)
+                all_numeric = all_ascii and all(x[i].isnumeric() for x in str_data)
+
+                if all_alpha:
+                    N = 26
+                elif all_numeric:
+                    N = 10
+
+                if all_alpha or all_numeric:
+                    S = st.entropy([*Counter(x[i] for x in str_data).values()])
+                    randomness_per_index.append(S / np.log(N))
+                else:
+                    break
+            else:
+                if np.mean(randomness_per_index) > 0.95:
+                    return True
+
```

Comment:
This approach seems overly complex to me. A safe assumption would be that we're bound never to have whitespaces (` `) in a foreign key and that the character density is tiny (e.g. < 50).

In that situation a better approach might be to check for the number of distinct values in the column itself, if that number is ~= the length of the data itself + there are no spaces (i.e. it's probably not text) => foreign key.

@paxcema and @maximlopin what are your thoughts on that ? We certainly need to find some solution here at any rate.

The alternative would be, based on what @paxcema  is saying, to allow for alphanumeric + number and have `N=36`, or even to allow for those + common id characters (e.g. `-` and `_`) and increase N to 38 or 40 or w/e.

Also, since we are now being much more aggressive with foreign key detection we should add a `force_use_column` argument to the `advanced_args` in case we ever mis-diagnose something as a foreign key.

Reply:
I think we already have logic for disabling handling foreign keys: https://github.com/mindsdb/mindsdb_native/blob/staging/mindsdb_native/libs/phases/type_deductor/type_deductor.py#L322

Reply:
> In that situation a better approach might be to check for the number of distinct values in the column itself, if that number is ~= the length of the data itself + there are no spaces (i.e. it's probably not text) => foreign key.

I like this idea, but I think it should be implemented in a separate PR

Reply:
I agree with the approach @George3d6 suggests, it feels intuitive as it's similar to the check I did when inspecting the column manually. In any case, this approach seems to work so we should leave the alternative for another PR, as Maxim says.